polynote-spark/src/main/scala/polynote/kernel/lang/sql/SparkSqlInterpreter.scala

```diff
@@ -0,0 +1,181 @@
+package polynote.kernel.lang.sql
+
+import java.io.File
+
+import cats.instances.list._
+import cats.effect.IO
+import cats.syntax.traverse._
+import cats.syntax.functor._
+import fs2.concurrent.Enqueue
+import org.apache.spark.sql.catalyst.parser.{SqlBaseBaseVisitor, SqlBaseParser}
+import org.apache.spark.sql.catalyst.parser.SqlBaseParser.SingleStatementContext
+import org.apache.spark.sql.{DataFrame, Dataset, SparkSession}
+import polynote.kernel._
+import polynote.kernel.lang.LanguageKernel
+import polynote.kernel.util.{Publish, RuntimeSymbolTable}
+import polynote.messages.{ShortString, TinyList, TinyString}
+
+import scala.collection.mutable
+
+class SparkSqlInterpreter(val symbolTable: RuntimeSymbolTable) extends LanguageKernel[IO] {
+
+  def predefCode: Option[String] = None
+
+  private val spark = SparkSession.builder().getOrCreate()
+  import spark.implicits._
+  private val parser = new Parser
+
+  private lazy val datasetType = symbolTable.global.typeOf[Dataset[Any]].typeConstructor
+
+  private def dataFrames[D <: Decl](symbols: Seq[D]) = symbols.collect {
+    case rv if rv.scalaType.dealiasWiden.typeConstructor <:< datasetType => rv
+  }.toList
+
+  private def registerTempViews(identifiers: List[Parser.TableIdentifier]) = {
+    val candidates = identifiers.collect {
+      case Parser.TableIdentifier(None, name) => name
+    }.toSet
+
+    dataFrames(symbolTable.currentTerms).collect {
+      case rv if candidates(rv.name.toString) =>
+        val nameString = rv.name.toString
+        IO.pure(rv.value.asInstanceOf[Dataset[_]]).flatMap(ds => IO(ds.createOrReplaceTempView(nameString)).as(nameString))
+    }
+  }
+
+  private def dropTempViews(views: List[String]) = views.map(name => IO(spark.catalog.dropTempView(name))).sequence.as(())
+
+  private def outputDataFrame(df: DataFrame, out: Enqueue[IO, Result]) = {
+    // TODO: We should have a way of just sending raw tabular data to the client for it to display and maybe plot!
+    //       But we'd have to write some actual content to the notebook file... maybe Output could take multiple
+    //       representations of the content and each representation could be possibly written to the notebook and
+    //       possibly sent to the client?
+    //       Sending raw table data was the reason for starting DataType stuff in polynote-runtime, but I haven't done
+    //       anything with it yet (JS)
```

Comment:
+1, this would be cool. We could look into what Jupyter/Zeppelin do for this. I suspect they both might save the HTML representation. 

I guess we might be worried if someone saves a huge 10k line table... for that matter, how do plots get saved in the json?